teacher.rb

```diff
@@ -0,0 +1,14 @@
+class Teacher < Person
+  attr_reader :id
+  attr_accessor :specialization
+
+  def initialize(specialization)
+    super()
+    @id = Random.rand(1..1000)
```

Comment:
[OPTIONAL] - Since the Teacher is inheriting from the `Person`, you really do not need to add the `id` here again.